src/zenkit/core/vec2f.py

```diff
@@ -0,0 +1,231 @@
+import math
+from ctypes import Structure, c_float
+from typing import ClassVar, Any
+
+class Vec2f(Structure):
+    """
+    A class to represent a 2D vector using floats, supporting various vector operations.
+    
+    Attributes:
+        x (float): The x-component of the vector.
+        y (float): The y-component of the vector.
+    """
+
+    _fields_: ClassVar[tuple[str, Any]] = [
+        ("x", c_float),
+        ("y", c_float),
+    ]
+
+    def __repr__(self) -> str:
+        """
+        Return a string representation of the vector.
+
+        Returns:
+            str: A string in the format 'Vec2f(x, y)'.
+        """
+        return f"Vec2f(x={self.x}, y={self.y})"
+    
+    def __neg__(self):
+        """
+        Negate the vector (reverse the direction of the vector).
+
+        Returns:
+            Vec2f: A new Vec2f instance with negated components.
+        """
+        return Vec2f(-self.x, -self.y)
+    
+    def __eq__(self, other):
+        """
+        Check if two vectors are equal component-wise.
+
+        Args:
+            other (Vec2f): The vector to compare with this vector.
+
+        Returns:
+            bool: True if the vectors are equal, False otherwise.
+
+        Raises:
+            TypeError: If the comparison is not between two Vec2f objects.
+        """
+        if isinstance(other, Vec2f):
+            return self.x == other.x and self.y == other.y
+        raise TypeError("Comparison is only supported between two Vec2f objects")
```

Comment:
`__eq__` should return `False` instead of throwing when comparing against an object of another type.